src/lib/template-result.ts

```diff
@@ -39,32 +39,29 @@ export class TemplateResult {
    * Returns a string of HTML used to create a `<template>` element.
    */
   getHTML(): string {
-    const l = this.strings.length - 1;
-    let html = '';
-    let isTextBinding = true;
-    for (let i = 0; i < l; i++) {
-      const s = this.strings[i];
-      html += s;
-      const close = s.lastIndexOf('>');
-      // We're in a text position if the previous string closed its last tag, an
-      // attribute position if the string opened an unclosed tag, and unchanged
-      // if the string had no brackets at all:
-      //
-      // "...>...": text position. open === -1, close > -1
-      // "...<...": attribute position. open > -1
-      // "...": no change. open === -1, close === -1
-      isTextBinding =
-          (close > -1 || isTextBinding) && s.indexOf('<', close + 1) === -1;
-
-      if (!isTextBinding && rewritesStyleAttribute) {
-        html = html.replace(lastAttributeNameRegex, (match, p1, p2, p3) => {
-          return (p2 === 'style') ? `${p1}style$${p3}` : match;
-        });
+    const endIndex = this.strings.length - 1;
+    return this.strings.reduce((acc, s, i) => {
+      if (i === endIndex) {
+        return acc + s;
+      }
+      // Append a suffix to all bound attribute names to opt out of special
+      // attribute value parsing that IE11 and Edge do, like for style and
+      // many SVG attributes.
+      // The Template class also appends the same suffix when looking up
+      // attributes to creat Parts.
+      let addedMarker = false;
+      acc += s.replace(lastAttributeNameRegex, (match, p1, p2, p3) => {
+        addedMarker = true;
+        return (p2.substring(p2.length - boundAttributeSuffix.length) ===
+                boundAttributeSuffix) ?
+            match :
+            `${p1}${p2}${boundAttributeSuffix}${p3}${marker}`;
```

Comment:
That happens with multiple bindings per attribute. But I did change this replace to run on the individual string rather than the whole template up to this point (which should be a nice perf win on large templates), so maybe this isn't needed.

Reply:
Current tests pass without that check, which makes sense.

Reply:
Ok I see now; so I just confirmed the current change will result in:
```js
html`<div style="color: ${'red'}; background: ${'blue'}"></div>`
```
ending up as this (first gets a `marker`, second gets a `nodeMarker`):
```
<div style$lit$="color: {{lit-...}}; background: <!--{{lit-...}}-->"></div>
```
Is that ok?

I forget the reason we don't just use `nodeMarker` in attributes, seems like it would work ok, since it's switched to attribute parsing at that point anyway, which doesn't care about HTML comments.  If so, seems like we could just get rid of the different marker types and make the `markerRegex` simpler too.

Reply:
We don't use a nodeMark all the time because in unquoted attributes is abruptly ends the tag. It's ok to use it in quoted attributes though. There's an existing test with some markup in an attribute and a binding after it that triggers the nodeMarker, but it's ok because you can only have that situation with a quoted attributes. Same here. You can only have multiple bindings in a quoted attribute.

Reply:
Ok got it, that's tricky -- probably warrants a comment here near the regex (e.g. "The regex both prefixes attribute names, and also adds an unquoted-attribute-safe marker for the first expression in an attribute. Note, subsequent attribute expressions will use node markers, and this is safe since attributes with multiple expressions are guaranteed to be quoted.").

[This comment](https://github.com/Polymer/lit-html/pull/640/files#diff-d48cd89dfde78b0aabf891be6a6e8a9fR24) could also be updated to indicate they are also used in compound attribute bindings in non-leading positions, which are guaranteed to be quoted.

Reply:
done